### ISSUE TITLE
Train  a SegFormer  on a Custom Dataset

### ISSUE THREAD

**Issue by jhayle21** [roboflow/notebooks#99]
### Search before asking

- [X] I have searched the Roboflow Notebooks [issues](https://github.com/roboflow-ai/notebooks/issues) and found no similar bug report.

### Notebook name

Roboflow How to Train SegFormer.ipynb

### Bug

![REDUCE CODE](https://user-images.githubusercontent.com/73093075/232224874-f326cb80-8a10-40e6-856d-90d91b74dc62.png)
![PL  TRAINER](https://user-images.githubusercontent.com/73093075/232224886-e0a41ea3-00ea-46db-b636-600df3af3f6b.png)

i just copy the code and try it but it wont work.

### Environment

-Google Colab

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by SkalskiP** [comment]
@Jacobsolawetz could you take a look at this bug report?

**Comment by mazatov** [comment]
I had the same error just now. Were you able to figure it out, @jhayle21 ?

**Comment by robmarkcole** [comment]
This is because pytorch-lightning dependency is not stated, and running now will install `'2.0.1.post0'` which has breaking changes from 1.X. Note there is also an attribute error from `feature_extractor.reduce_labels = False`
Therefore this notebook needs updating

**Comment by mazatov** [comment]
@robmarkcole,  Thank you 100% agree. It also applies to `test_epoch_end`  ---> `on_test_epoch_end`

**Comment by SkalskiP** [comment]
Hi @jhayle21, @mazatov and @robmarkcole 👋🏻! I just pushed a fixed version of our SegFormer notebook. It is not ideal, but it works as expected. What I did was:
- `pytorch-lightning<2.0.0` - fix the version of `PyTorch Lightning` to be the latest version below `2.0.0`. In the future, we need to update API used in the notebook. But for now, it should be fine.
- allow for latest version of `roboflow` to enable new authentication API
- update transformers API `reduce_labels` -> `do_reduce_labels`

I'm closing the issue but feel free to reopen it in the future if needed.

**Comment by SkalskiP** [comment]
Hi, @mazatov 👋🏻! 

This looks like a very interesting project. How large is the dataset? How long did you train?

**Comment by mazatov** [comment]
The train dataset is about 16,000 images. I trained it only for 10 epochs
hoping to see some progress before moving away from colab to something
where it won't get interrupted. I was surprised to see this type of output.
Usually other networks immediately give something quite reasonable very fast

**Comment by SkalskiP** [comment]
The main problem here is the fact that the objects you try to detect are thin. In the past, I trained two models like that. And I managed to do that with Mask R-CNN and YOLOv7.

**Comment by mazatov** [comment]
I've tried Yolov7/8 and they do alright. I was just hoping to get better results with segformer or mask2former.
